mergify/labels-copier/copy.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# Environment variables:
+#  PR_URL
+#  REPOSITORY_URL
+#  ADDITIONAL_LABELS
+#  EXCLUDED_LABEL
+#  DRY_RUN
+#  RUNNER_DEBUG
+#
+# You can run this locally for testing purposes:
+#
+# $ REPOSITORY_URL=https://github.com/elastic/apm-server PR_URL=https://github.com/elastic/apm-server/pull/15035 EXCLUDED_LABEL=backport-* DRY_RUN=true sh mergify/labels-copier/copy.sh
+#
+
+set -eo pipefail
+
+# Support debugging in GitHub actions when RUNNER_DEBUG is set.
+if [ -n "$RUNNER_DEBUG" ] ; then
+  set -x
+fi
+
+add_label() {
+  local label=$1
+  local pr_url=$2
+
+  if [ "$DRY_RUN" == "true" ]; then
+    echo ">> [dry-run]: $label will be added"
+  else
+    gh pr edit --add-label "$label" "$pr_url"
+  fi
+}
+
+
+pr_number=$(gh pr view --json body -q ".body" "$PR_URL" | sed -n -e '/automatic backport of pull request/,/done/p' | cut -d"#" -f2 | cut -d" " -f1)
+gh pr view --json labels -q '.labels[]|.name' ${REPOSITORY_URL}/pull/$pr_number | while read label ; do
+  if [[ -z "$labels" ]] || [[ ",$labels," =~ ",$label," ]]; then
+    if [[ -n "$EXCLUDED_LABEL" ]] && [[ $label =~ $EXCLUDED_LABEL ]]; then
+      echo ">> $label is excluded and will not be added since matches '$EXCLUDED_LABEL'"
+    else
+      add_label "$label" "$PR_URL"
+    fi
+  fi
+  for additional_label in $(echo $ADDITIONAL_LABELS | sed "s/,/ /g") ; do
+    add_label "$additional_label" "$PR_URL"
+  done
+done
```

Comment:
it is complex, you can use Action script to add the labels directly, duplicated label will not apply

Reply:
It's not that complex IMO. I can simplify it a bit.

And still, I see a much powerful feature to use a shell script, so I can test it out in isolation locally than using the `github-script` action step, it's hard to test it locally

Reply:
See [`9a9718a` (#207)](https://github.com/elastic/oblt-actions/pull/207/commits/9a9718a19b0de6ea0ad69f70609475c892eeece9)


<img width="1680" alt="image" src="https://github.com/user-attachments/assets/8ca66785-119a-490b-9ec6-9950b83085ff" />

<img width="553" alt="image" src="https://github.com/user-attachments/assets/4a545571-129e-4cd1-b05f-ca1f3e8d1546" />


https://github.com/elastic/apm-server/pull/15035#event-15865601616